dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java

```diff
@@ -304,6 +304,30 @@ public int deleteSoftDeletedTrackedEntityInstances()
         return result;
     }
 
+    @Override
+    public void prunePeriods()
```

Comment:
Consider adding a note in Javadoc that explains that this SQL query must be updated if new table with foreign key to the period table is added.